src/actions/promote.ts

```diff
@@ -41,4 +50,114 @@ export async function promote(ignoreSslErrors?: boolean) {
+        await checkManifestStability(kubectl, resourceTypes);
+        blueGreenDeploymentHelper.blueGreenRouteTraffic(kubectl, inputManifestFiles, blueGreenDeploymentHelper.NONE_LABEL_VALUE);
+        blueGreenDeploymentHelper.deleteOpposite(kubectl, inputManifestFiles, blueGreenDeploymentHelper.BLUE_GREEN_NEW_LABEL_VALUE);
+        blueGreenDeploymentHelper.cleanSetUpSMI(kubectl, inputManifestFiles);    
+    } else {
+        core.debug('deleting old deployment and making new ones');
+        let result = await blueGreenDeploymentHelper.blueGreenPromote(kubectl, inputManifestFiles);
+        const deployedManifestFiles = result.newFilePaths;
+        const resourceTypes: Resource[] = KubernetesObjectUtility.getResources(deployedManifestFiles, models.deploymentTypes.concat([models.DiscoveryAndLoadBalancerResource.service]));
+        await checkManifestStability(kubectl, resourceTypes);
+        core.debug('routing to new deployments');
+        blueGreenDeploymentHelper.blueGreenRouteService(kubectl, inputManifestFiles, blueGreenDeploymentHelper.NONE_LABEL_VALUE);
+        blueGreenDeploymentHelper.deleteOpposite(kubectl, inputManifestFiles, blueGreenDeploymentHelper.BLUE_GREEN_NEW_LABEL_VALUE);
+    }
+}
+
+async function checkManifestStability(kubectl: Kubectl, resources: Resource[]): Promise<void> {
+    await KubernetesManifestUtility.checkManifestStability(kubectl, resources);
+}
+
+
+function getManifestFiles(manifestFilePaths: string[]): string[] {
+    const files: string[] = utils.getManifestFiles(manifestFilePaths);
+
+    if (files == null || files.length === 0) {
+        throw new Error(`ManifestFileNotFound : ${manifestFilePaths}`);
+    }
+
+    return files;
+}
+
+function updateContainerImagesInManifestFiles(filePaths: string[], containers: string[]): string[] {
+    if (!!containers && containers.length > 0) {
+        const newFilePaths = [];
+        const tempDirectory = fileHelper.getTempDirectory();
+        filePaths.forEach((filePath: string) => {
+            let contents = fs.readFileSync(filePath).toString();
+            containers.forEach((container: string) => {
+                let imageName = container.split(':')[0];
+                if (imageName.indexOf('@') > 0) {
+                    imageName = imageName.split('@')[0];
+                }
+                if (contents.indexOf(imageName) > 0) {
+                    contents = utils.substituteImageNameInSpecFile(contents, imageName, container);
+                }
+            });
+
+            const fileName = path.join(tempDirectory, path.basename(filePath));
+            fs.writeFileSync(
+                path.join(fileName),
+                contents
+            );
+            newFilePaths.push(fileName);
+        });
+
+        return newFilePaths;
+    }
+
+    return filePaths;
+}
+

```

Comment:
Does this do anything different from the common one? 
You could have called them using different inputs right?